### ISSUE TITLE
Language pack inheritance order is incorrect

### ISSUE THREAD

**Issue by koenner01** [magento/magento2#8691]
Direct quote from the Magento 2 documentation:
(http://devdocs.magento.com/guides/v2.0/config-guide/cli/config-cli-subcommands-i18n.html)

> If necessary, you can specify several parent packages. The parent packages are applied on a first listed, first used basis.

And 

> In the preceding example:
> 
> language_package_one descends from en_au_package and en_au_package descends from en_ie_package
> language_package_two descends from en_ca_package and en_ca_package descends from en_us_package
> If the Magento application cannot find word or phrase in the en_GB package, it looks in other packages in following sequence:
> 
> parent-package-one/language_package_one
> <vendorname>/en_au_package
> <vendorname>/en_ie_package
> parent-package-two/language_package_two
> <vendorname>/en_ca_package
> <vendorname>/en_us_package
> Specifying all inheritances between the language packages might result in creating circular inheritance chains. Use Magento\Test\Integrity\App\Language\CircularDependencyTest test to locate and fix such chains.
> 

The documentation clearly states that 'package_one' should be loaded before 'package_two'; However if your 'package_one' name comes after 'package_two' alphabetically the order will not be respected.

### Preconditions
1. Magento 2.4-develop
2. PHP7.2

### Steps to reproduce
1. Create 2 language packs with package names "pack_a" & "pack_z"
2. Create a project language pack which uses the previously created packs.
3. Make sure you add "pack_z" as the first package to use then "pack_a"

```
<?xml version="1.0"?>
<language xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:App/Language/package.xsd">
   ...
    <use vendor="myvendor" package="pack_z"/>
    <use vendor="myvendor" package="pack_a"/>
</language>
```

### Expected result
1. Translations should be firstly fetched from "pack_z" then "pack_a"

### Actual result
1. Translations are firstly fetched from "pack_a" then from "pack_z"

**Comment by magento-engcom-team** [comment]
@koenner01, thank you for your report.
We've created internal ticket(s) MAGETWO-83802 to track progress on the issue.

**Comment by magento-engcom-team** [comment]
Hi @milindsingh. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by engcom-Alfa** [comment]
The issue is reproduced on a 2.4-develop instance

**Actual Result:**
![screen](https://user-images.githubusercontent.com/51679138/72734214-eff7f680-3ba1-11ea-9c64-c1dd6549e9b2.png)

**Comment by sergiy-v** [comment]
I'm working on the issue.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Alfa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-30628` were created

**Issue Available**: @engcom-Alfa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by slavvka** [comment]
Hi @koenner01. Thank you for your report. 
The issue has been fixed in magento/magento2#26420 by @sergiy-v in 2.4-develop branch
Related commit(s):
 - [8ce08227b0cc10f16c3d59a25c432a615cb1864b](https://github.com/magento/magento2/commit/8ce08227b0cc10f16c3d59a25c432a615cb1864b)
 - [51ddad31d70e27df288c37fd566e628eadacf45f](https://github.com/magento/magento2/commit/51ddad31d70e27df288c37fd566e628eadacf45f)
 - [0237c023976105482365430961559a289cd9aaee](https://github.com/magento/magento2/commit/0237c023976105482365430961559a289cd9aaee)
 - [c42ebc1c298c6262bbdf35e0a8682b693b8186d9](https://github.com/magento/magento2/commit/c42ebc1c298c6262bbdf35e0a8682b693b8186d9)

The fix will be available with the upcoming 2.4.0 release.
